homeassistant/components/motion_blinds/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""The motion_blinds component."""
+from asyncio import TimeoutError as AsyncioTimeoutError
+from datetime import timedelta
+import logging
+from socket import timeout
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_GATEWAY, MANUFACTURER
+from .gateway import ConnectMotionGateway
+
+_LOGGER = logging.getLogger(__name__)
+
+MOTION_PLATFORMS = ["cover", "sensor"]
+
+
+async def async_setup(hass: core.HomeAssistant, config: dict):
+    """Set up the Motion Blinds component."""
+    return True
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up the motion_blinds components from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+    key = entry.data[CONF_API_KEY]
+
+    # Connect to motion gateway
+    connect_gateway_class = ConnectMotionGateway(hass)
+    if not await connect_gateway_class.async_connect_gateway(host, key):
```

Comment:
Can this throw an exception?

Reply:
No in principle not, the async_connect_gateway has a try-catch block that catches timeouts, other exeptions schould not be possible to my knoladge.